### DISCUSSION TITLE
"Only INSERT" RLS policy for public data base doesn't work

### DISCUSSION THREAD

**Discussion by LeoHeuser** [supabase/supabase#4107] [Questions] [answered]
Hey guys 👋🏼,
I created up a public Supabase db where I want to store anonymous feedback from a iOS (Swift and SwiftUI) app. The user provides three String parameters to the db (Email, kind of feedback and the feedback text). In the app I use the _anon key_ to contact the API.

When the RLS is off it works just fine. But then everyone with the key can see (SELECT) the feedback I've got so far. Therefore I want to limit the access to the database to "insert only" for now.
My try for now was: 
`CREATE POLICY "Everyone can only INSERT feedback" ON public.feedback_db FOR INSERT WITH CHECK (true);` - pretty basic

The Xcode console return (with RLS on):
> new row violates row-level security policy for table "feedback_db"

I also tried to send an identifier via the app, so that it is obvious it is from the app.
Nothing worked so far. Hopefully one of you can help me out 🧐

Best,
Leo 👋🏼

**Comment by GaryAustin1** [comment, marked as answer]
Supabase JS does a select on inserts, I'm not sure about other libraries.  For JS you can turn off that select:
https://supabase.com/docs/reference/javascript/insert

**Reply by LeoHeuser** [reply]
This looks like a pretty good hint. Thanks @GaryAustin1! I also looked it up also in the discussions here and I need to communicate the `returning=minimal` command.

I tried a couple of thinks with Swift but I didn‘t get it. Guess I need to find someone who knows the trick 🤓

**Reply by GaryAustin1** [reply]
You will likely need to start an issue here:  https://github.com/supabase-community/postgrest-swift/issues

I glanced at the swift code and it does not look like they added the support for making return type an option like they have in the javascript code.  The below (from the Supabase swift code) sets return to representation with no  option.

public func insert(values: Any, upsert: Bool = false, onConflict: String? = nil) -> PostgrestBuilder {
        method = "POST"
        headers["Prefer"] = upsert ? "return=representation,resolution=merge-duplicates" : "return=representation"

You could always make the call directly to the rest/api for that insert using whatever is fetch in swift or do the insert in an rpc function....
